[kv] coordinator must handle inconsistency or broken snapshot scenarios

platinumhamburg opened the pull request:
<!--
*Thank you very much for contributing to Fluss - we are happy that you want to help us improve Fluss. To help the community review your contribution in the best possible way, please go through the checklist below, which will get the contribution into a shape in which it can be best reviewed.*

## Contribution Checklist

  - Make sure that the pull request corresponds to a [GitHub issue](https://github.com/alibaba/fluss/issues). Exceptions are made for typos in JavaDoc or documentation files, which need no issue.

  - Name the pull request in the format "[component] Title of the pull request", where *[component]* should be replaced by the name of the component being changed. Typically, this corresponds to the component label assigned to the issue (e.g., [kv], [log], [client], [flink]). Skip *[component]* if you are unsure about which is the best component.

  - Fill out the template below to describe the changes contributed by the pull request. That will give reviewers the context they need to do the review.

  - Make sure that the change passes the automated tests, i.e., `mvn clean verify` passes.

  - Each pull request should address only one issue, not mix up code from multiple issues.


**(The sections below can be removed for hotfixes or typos)**
-->

### Purpose

<!-- Linking this pull request to the issue -->
Linked issue: close #1469 

<!-- What is the purpose of the change -->

### Brief change log

<!-- Please describe the changes made in this pull request and explain how they address the issue -->

### Tests

<!-- List UT and IT cases to verify this change -->

### API and Format

<!-- Does this change affect API or storage format -->

### Documentation

<!-- Does this change introduce a new feature -->

wuchong reviewed (commented):
Could you add a test to reproduce this problem and verify the bug is fixed? Maybe a unit test in `CompletedSnapshotStoreManagerTest`.

platinumhamburg commented:
> Could you add a test to reproduce this problem and verify the bug is fixed? Maybe a unit test in `CompletedSnapshotStoreManagerTest`.

Thank you for your feedback. I have addressed all of your suggestions and supplemented the test cases to ensure correctness.

wuchong commented on fluss-server/src/test/java/com/alibaba/fluss/server/coordinator/CompletedSnapshotStoreManagerTest.java:
```diff
@@ -172,6 +181,50 @@ void testRemoveCompletedSnapshotStoreFromManager() throws Exception {
+        TableBucket tableBucket = new TableBucket(1, 1);
+        CompletedSnapshot validSnapshot =
+                KvTestUtils.mockCompletedSnapshot(tempDir, tableBucket, 1L);
+        CompletedSnapshotHandle validSnapshotHandle = mock(CompletedSnapshotHandle.class);
+        when(validSnapshotHandle.getSnapshotId()).thenReturn(validSnapshot.getSnapshotID());
+        when(validSnapshotHandle.getMetadataFilePath())
+                .thenReturn(validSnapshot.getMetadataFilePath());
+        when(validSnapshotHandle.retrieveCompleteSnapshot()).thenReturn(validSnapshot);
+
+        CompletedSnapshot invalidSnapshot =
+                KvTestUtils.mockCompletedSnapshot(tempDir, tableBucket, 2L);
+        CompletedSnapshotHandle invalidSnapshotHandle = mock(CompletedSnapshotHandle.class);
+        when(invalidSnapshotHandle.getSnapshotId()).thenReturn(invalidSnapshot.getSnapshotID());
+        when(invalidSnapshotHandle.getMetadataFilePath())
+                .thenReturn(invalidSnapshot.getMetadataFilePath());
```
Please avoid use Mockito, use reusable test implementations instead. (see [the code quality guide and reason here](https://flink.apache.org/how-to-contribute/code-style-and-quality-common/#avoid-mockito---use-reusable-test-implementations)). 

Here, we can easily construct the valid and invalid `CompletedSnapshotHandle` using its constructor, one has valid path and the other has invlid path. This makes the code easy to maintain for the future, and cover more code path.

platinumhamburg replied:
Thanks for the suggestion. I've refactored this case to avoid using Mockito.

wuchong reviewed (commented):
Thanks @platinumhamburg for the fixing. 

Just a quick note: could you please avoid squashing commits while responding to review comments? Keeping the changes in separate commits makes it much easier to track what has been updated since the last review.

When commits are squashed, it becomes difficult to distinguish new changes from previous ones, forcing reviewers to review the entire changes again. This significantly increases review time and may slow down the overall merge process.

The committer will help to squash commits and improve the commit message before merging the PR. So don’t worry about having multiple "fix" commits in PR.
